src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -199,14 +200,15 @@ public String getDisplayName() {
      */
     @RequirePOST
     public HttpResponse doSubmit(StaplerRequest request) throws IOException, ServletException, InterruptedException {
+        Run<?, ?> run = getRun();
         if (request.getParameter("proceed")!=null) {
             doProceed(request);
         } else {
             doAbort();
         }
 
         // go back to the Run console page
-        return HttpResponses.redirectTo("../../console");
+        return HttpResponses.redirectTo(ConsoleURLProvider.get().getConsoleURL(run));
```

Comment:
Hmm, thinking about this, I wonder if we should encourage plugins like this to use `ConsoleURLProvider` or `DisplayURLProvider`. `ConsoleURLProvider` is only really needed for links in Jenkins core itself. Plugins could use `DisplayURLProvider.getConsoleURL` right away if we split the `display-url-api` PR into two parts, one just for `DisplayURLProvider.getConsoleURL` and one to update the core dependency and implement `ConsoleURLProvider`.

Reply:
If `display-url-api` is not installed, the `DisplayURLProvider` is not available from the current plugin, where `ConsoleURLProvider` is.

Reply:
Yes, but I think it should not be a big deal to add `display-url-api` as a compile-scope dependency to plugins that produce console links.

Reply:
Outdated based on upstream changes, see https://github.com/jenkinsci/pipeline-input-step-plugin/pull/145/commits/fd34d23a733c82aef6f80f52d511d945d0ca6f9d.